src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2376,7 +2459,72 @@ def _reindex_axis_0(
                             index_column_snowflake_quoted_identifiers=new_qc._modin_frame.index_column_snowflake_quoted_identifiers,
                         )
                     )
-
+        if is_index:
+            modin_frame = new_qc._modin_frame
+            # We need to get the new quoted identifier after filling happens.
+            row_position_column = modin_frame.data_column_snowflake_quoted_identifiers[
+                modin_frame.data_column_pandas_labels.index("indices")
```

Comment:
Again, I'd prefer to access `"indices"` by its position (is that `data_column_pandas_labels[-1]`?) rather than by value.

Reply:
Resolved [here](https://github.com/snowflakedb/snowpark-python/pull/1935/files#r1687118143).